src/polyfill.rs

```diff
@@ -45,6 +45,14 @@ pub mod slice {
         u32::from(buffer[3])
     }
 
+    #[inline(always)]
+    pub fn u8_as_be_u32(dest: &mut [u8], value: u32) {
```

Comment:
This should be named `be_u8_from_u32` to be consistent with `u32_from_be_u8`.

Instead of taking a `dest: &mut [u8]` parameter, you can return a `[u8; 4]` result.